lib/ansible/modules/cloud/hcloud/hcloud_network.py

```diff
@@ -197,6 +218,7 @@ def define_module():
                 name={"type": "str"},
                 ip_range={"type": "str"},
                 labels={"type": "dict"},
+                delete_protection={"type": "bool", "default": False},
```

Comment:
the default should be `None` (remove the default, also in doc) because otherwise the setting would be updated to false if a user doesn't specify it. A extend test case would be

```yaml
- name: test update Network delete protection idempotency
  hcloud_network:
    name: "{{hcloud_network_name}}"
    ip_range: "10.0.0.0/8"
    delete_protection: true
  register: network
- name: verify test update Network delete protection idempotency
  assert:
    that:
      - network is not changed
      - network.hcloud_network.delete_protection is sameas true

- name: test Network without delete protection set to be idempotent
  hcloud_network:
    name: "{{hcloud_network_name}}"
    ip_range: "10.0.0.0/8"
  register: network
- name: verify test Network without delete protection set to be idempotent
  assert:
    that:
      - network is not changed
      - network.hcloud_network.delete_protection is sameas true
```